### PR TITLE
JSON serialization of circuits

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
Signed-off-by: Mihai Budiu <mbudiu@vmware.com>
@Kixiron hopefully this can be used as a basis for the JIT. The output is on the large side, and the compiler does not yet have a flag to output json.

### CODE REVIEW

File: SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/ToJSONVisitor.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+public class ToJSONVisitor extends CircuitVisitor {
+    protected final IndentStream builder;
+
+    public ToJSONVisitor(IndentStream builder) {
+        super(true);
+        this.builder = builder;
+    }
+
+    @Override
+    public boolean preorder(DBSPCircuit circuit) {
```
Comment (lalithsuresh) [inline SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/ToJSONVisitor.java:45]: You could use org.json instead of reinventing all this.

Comment (mbudiu-vmw) [reply SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/ToJSONVisitor.java:45]: Jackson is doing the heavy lifting

Comment (lalithsuresh) [reply SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/ToJSONVisitor.java:45]: No, I mean all the code to do pretty printing and produce the json string could be delegated to Jackson or org.json.

**Comment by mbudiu-vmw** [comment]
With this last PR you can dump json from the SQL compiler as follows:
```
cd SQL-compiler
./sql-to-dbsp -j -o x.json file.sql
```

**Pull request merged**